pkg/operation/botanist/health_check.go

```diff
@@ -683,12 +683,25 @@ func (b *Botanist) checkSystemComponents(
 	if exitCondition := checker.CheckExtensionCondition(condition, extensionConditions); exitCondition != nil {
 		return exitCondition, nil
 	}
-	if established, err := b.CheckVPNConnection(ctx, logrus.NewEntry(logger.NewNopLogger())); err != nil || !established {
-		msg := "VPN connection has not been established"
+
+	var tunnelName common.TunnelName
+	// If kubernetes version is >= 1.18 deploy the proxy-agent, otherwise deploy the proxy server
+	kubernetesVersionGeq118, err := versionutils.CheckVersionMeetsConstraint(b.Shoot.Info.Spec.Kubernetes.Version, ">= 1.18")
```

Comment:
You also need to check the Gardener version here. Concretely, assuming this PR will be released as part of Gardener `x.y.z` then you'd only execute the konnectivity check if the shoot was already reconciled by Gardener `x.y.z` and if >= 1.18.

Reply:
I am not sure I get this comment, why is the kubernetes version not enough given this PR current state, in this case if its > 1.18 we assume that konnectivity it is deployed.

Reply:
Imagine this feature being rolled out with Gardener `x.y.z`. If reconciliation of shoots only happens in maintenance time window then all shoots are only reconciled with a version < `x.y.z`, hence, still have the vpn-shoot. Your health check will always report an error until the clusters are reconciled the first time now. To avoid this, you have to check if the shoot was already reconciled with `x.y.z`. Only then you can perform the konnectivity check